examples/api/test/widgets/focus_scope/focus.2_test.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/focus_scope/focus.2.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('Adds children through button', (WidgetTester tester) async {
+    await tester.pumpWidget(const example.FocusExampleApp());
+    expect(find.byIcon(Icons.add), findsOneWidget);
+    expect(find.text('CHILD 0'), findsOneWidget);
+
+    for (int i = 1; i <= 20; i += 1) {
+      await tester.tap(find.byIcon(Icons.add));
+      await tester.pumpAndSettle();
+      expect(find.text('CHILD $i'), findsOneWidget);
+      expect(find.textContaining('CHILD '), findsNWidgets(i + 1));
+    }
+  });
+
+  testWidgets('Inserts focus nodes', (WidgetTester tester) async {
+    await tester.pumpWidget(const example.FocusExampleApp());
+    expect(find.byIcon(Icons.add), findsOneWidget);
+
+    for (int i = 0; i <= 10; i += 1) {
+      await tester.pumpAndSettle();
```

Comment:
This seems to weird to call right at the beginning of for loop?

Is this suppose to trigger frame update after `find.byIcon(Icons.add)` at the end of the loop? If yes, please move this below the tap action.

Reply:
Thanks for pointing this out. I'm not sure what my intentions in splitting them were. I went with your suggestion, but this will still have an unused tap and pump at the end.